tests/test_api.py

```diff
@@ -393,59 +384,44 @@ async def test_api_update_resource_url_since_load_catalog(
     # It does not create any duplicated resource.
     # The existing entry get updated accordingly.
 
-    # TODO: can be removed once we don't use legacy route anymore:
-    client_http_method: Callable = getattr(client, route["method"])
-
     # Test API call success
-    resp = await client_http_method(path=route["url"], headers=api_headers, json=payload)
+    resp = await client.put(path=f"/api/resources/{RESOURCE_ID}", headers=api_headers, json=payload)
     assert resp.status == 200
 
     res = await db.fetch(f"SELECT * FROM catalog WHERE resource_id = '{RESOURCE_ID}'")
     assert len(res) == 1
     res[0]["url"] == "https://example.com/resource-1"
 
 
-@pytest.mark.parametrize(
-    "route",
-    [
-        {"url": "/api/resources/", "method": "delete"},
-        {"url": "/api/resource/deleted/", "method": "post"},  # legacy route
-    ],
-)  # TODO: can be removed once we don't use legacy route anymore
-async def test_api_delete_resource(client, route, api_headers, api_headers_wrong_token):
-    # TODO: can be removed once we don't use legacy route anymore
-    client_http_method: Callable = getattr(client, route["method"])
-
-    # Test invalid DELETE data
-    stupid_delete_data: dict = {"stupid": "stupid"}
-    resp = await client_http_method(path=route["url"], headers=api_headers, json=stupid_delete_data)
+async def test_api_delete_resource(client, api_headers, api_headers_wrong_token):
+    # Test invalid resource_id
+    resp = await client.delete(path="/api/resources/STUPID-ID", headers=api_headers)
```

Comment:
I'm assuming this PR will need https://github.com/datagouv/data.gouv.fr/issues/1426 merged first, and this PR actually removes the legacy routes? (and thus removes the legacy test code).

Reply:
No, this PR keeps the legacy route (see https://github.com/datagouv/hydra/pull/132/files#diff-fafc7cf4d53a0c0beabd06c3a7dc5b6b5343e69d020f399c591658ca9cf8f150 and https://github.com/datagouv/hydra/pull/132/files#diff-22420b21d89ad900912688af4fed0206b66f6789c8438218eed67f8871cf0865 for the related tests)